caddyconfig/caddyfile/dispenser.go

```diff
@@ -201,6 +202,43 @@ func (d *Dispenser) Val() string {
 	return d.tokens[d.cursor].Text
 }
 
+// RawVal gets the raw text of the current token (including quotes).
+// If there is no token loaded, it returns empty string.
+func (d *Dispenser) RawVal() string {
+	if d.cursor < 0 || d.cursor >= len(d.tokens) {
+		return ""
+	}
+	quote := d.tokens[d.cursor].wasQuoted
+	if quote > 0 {
+		return string(quote) + d.tokens[d.cursor].Text + string(quote)
+	}
+	return d.tokens[d.cursor].Text
+}
+
+// ScalarVal gets value of the current token, converted to the
+// appropriate scalar type. If there is no token loaded, it returns nil.
+func (d *Dispenser) ScalarVal() interface{} {
+	if d.cursor < 0 || d.cursor >= len(d.tokens) {
+		return nil
+	}
+	quote := d.tokens[d.cursor].wasQuoted
+	text := d.tokens[d.cursor].Text
+
+	if quote > 0 {
```

Comment:
```suggestion
	if quote > 0 {
		// string literal
```
Just for people like me who get confused easily

Reply:
Should we add a method to Token like `WasQuoted() bool` ?

Reply:
I was actually thinking of something like `quote()` which returns the quote rune (or 0 if unquoted).

Reply:
Hmm, yeah, I guess if we change it to 0 then it would be falsey, right? So we wouldn't need `> 0`, so we probably don't need a method for it.

Reply:
Sort of; Go is strongly typed. But yeah, it's conventional to check for the "zero value" to see if something is set. Since 0 is technically a valid rune (just as 0 is a valid integer), it can be hard to differentiate zero value from unset (which is why many code bases use pointers in those cases, since they're nil-able), but in our case 0 is not a valid quote rune so we know that 0 just means "not quoted".